schema-engine/core/src/commands/diff.rs

```diff
@@ -132,38 +126,27 @@ async fn json_rpc_diff_target_to_connector(
     namespaces: Option<Namespaces>,
     preview_features: BitFlags<psl::PreviewFeature>,
 ) -> CoreResult<Option<(Box<dyn SchemaConnector>, DatabaseSchema)>> {
-    let read_prisma_schema_from_path = |schema_path: &str| -> CoreResult<String> {
-        std::fs::read_to_string(schema_path).map_err(|err| {
-            ConnectorError::from_source_with_context(
-                err,
-                format!("Error trying to read Prisma schema file at `{schema_path}`.").into_boxed_str(),
-            )
-        })
-    };
-
     match target {
         DiffTarget::Empty => Ok(None),
-        DiffTarget::SchemaDatasource(SchemaContainer { schema }) => {
-            let schema_contents = read_prisma_schema_from_path(schema)?;
-            let schema_dir = std::path::Path::new(schema).parent();
-            let mut connector = crate::schema_to_connector(&schema_contents, schema_dir)?;
+        DiffTarget::SchemaDatasource(schemas) => {
+            let schema_dir = crate::find_common_root_path(schemas);
+            let schema_dir = schema_dir.as_deref().and_then(|p| p.parent());
+            let sources: Vec<_> = schemas.to_psl_input();
```

Comment:
Make it so that the client can pass the config dir. Make it optional _if_ it's painful for testing